sgqlc/endpoint/http.py

```diff
@@ -163,9 +163,10 @@ def __call__(self, query, variables=None, operation_name=None,
         if extra_headers:
             headers.update(extra_headers)
 
-        headers.update({
-            'Accept': headers.get('Accept', 'application/json; charset=utf-8'),
-        })
+        if 'Accept' not in headers:
+            headers.update({
+                'Accept': headers.get('Accept', 'application/json; charset=utf-8'),
```

Comment:
hum... are you sure this wasn't working? I can't test right now, but the old code reads:

```py
headers.get('Accept', 'application/json; charset=utf-8'),
``` 

Which is basically "give me the existing `Accept` header, if it's not there, then use `application/json; charset=utf-8` as fallback."

If I'm not mistaken, this would set the `headers['Accept']` to its own value, in case it's already set. Your new could would even be rewritten as:

```py
        if 'Accept' not in headers:
            headers['Accept'] = 'application/json; charset=utf-8'
```

as `headers.get('Accept', 'application/json; charset=utf-8')` would always return `application/json; charset=utf-8` given the newly introduced `if`

Reply:
Yes you are correct. it should be the latter to only set.  However as you say it should only override it if it is not already set.

Reply:
I suspect this was due to updating the header whilst running the check for the Accept header could be causing the issue.

Reply:
it shouldn't as it first evaluate the `.get()` to later do the `.update()`.

That said, the new function is correct and maybe cleaner, so it can go in. I'm just trying to understand why it happened.